types/mdx/types.d.ts

```diff
@@ -62,3 +78,5 @@ export interface MDXModule {
      */
     default: MDXContent;
 }
+
+export {};
```

Comment:
Maybe document that this is needed for TS to see the file as a module?

Reply:
This is of course a breaking change; is anyone referencing the types that will now go away?

Reply:
Good catch. The implicitly exported types were never intended to be exported. Should I export them explicitly, and mark them as deprecated instead?

Reply:
That's up to you; I'm not an MDX types expert 😅 

I'm also wondering whether or not these types were secretly globals before without needing import but I think not?

Reply:
I just double checked and thankfully it’s not that bad.

@wooorm @ChristianMurphy WDYT? Should those previous accidental exports be explicitly exported now? It’s about `FunctionComponent`, `ClassComponent`, `Component`, and `NestedMDXComponents`.

If users did rely on this, the fix for them would be to import from `FunctionComponent` / `ClassComponent` / `Component` from their framework, probably React, which even uses the same names. I doubt anyone uses `NestedMDXComponents` type.

Reply:
I'd be okay with them being exported.
I could see them rarely being used by adopters working close to the internals of the project.
That said that use case is rare, excluding these from the types would probably work well for most.
And it may be better to wait and see how and why people would want access to these internals rather than immediately putting all of them to public.

Reply:
Yep, I'd hide them for now. If people use them, we can discuss their use cases and improve and expose from there

Reply:
Alright, then I’ll release this as-is :tada: